explorer_frontend/src/features/contracts/init.ts

```diff
@@ -651,3 +651,16 @@ sample({
   filter: $activeAppWithState.map((app) => !!app && !app.address),
   target: triggerShardIdValidation,
 });
+
+$shardsAmount.on(setShardsAmount, (_, shardsAmount) => shardsAmount);
```

Comment:
it's better here to subscribe on effectResult. Event should be used more for user interaction side.

Reply:
Done!